homeassistant/components/homepilot/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""The Rademacher HomePilot integration."""
+
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyhomepilot.api import HomePilotAPI
+from pyhomepilot.auth import Auth
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, SCAN_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["cover"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Rademacher HomePilot from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    session = aiohttp_client.async_create_clientsession(hass)
+    auth = Auth(session, entry.data[CONF_HOST], entry.data.get(CONF_PASSWORD))
+
+    if entry.data.get(CONF_PASSWORD) is not None:
+        await auth.async_login()
+
+    api = HomePilotAPI(auth)
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            return {device.uid: device for device in await api.async_get_all_devices()}
+        except Exception as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="homepilot",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=SCAN_INTERVAL),
+    )
+
+    # Fetch initial data
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data[DOMAIN][entry.entry_id] = {"api": api, "coordinator": coordinator}
```

Comment:
```suggestion
    hass.data[DOMAIN][entry.entry_id] = {DATA_API: api, DATA_COORDINATOR: coordinator}
```
define such kind of constants for this keys, since they are used multiple times in different places.